src/TransactionBuilder.cs

```diff
@@ -654,7 +654,8 @@ private LineItemModel GetMostRecentLine(string memberName = "")
         /// <returns></returns>
         public async Task<TransactionModel> CreateAsync()
         {
-            return await _client.CreateTransactionAsync(null, _model);
+            return await _client.CreateTransactionAsync(null, _model)
+                .ConfigureAwait(false);
```

Comment:
While adding a `ConfigureAwait()` throughout the codebase is a first step, it is not a complete solution for the possibility of a deadlock. In addition to adding the 'ConfigureAwait()' I also removed all blocking that was being done with the EAP pattern (async/await). If you take a look at the "The Blocking Hack" section of https://msdn.microsoft.com/en-us/magazine/mt238404 it gives an explanation as to why calling `ConfigureAwait()` might not be enough. While the SDK is not using WebClient like the articles example, I dont know if there are any other .Net APIs being use that always capture the context. Because of this uncertainty I took the approach of replacing all blocking EAP calls with a blocking APM call.